### PR TITLE
fix: dont OOM deleting large teams/organizations

### PR DESCRIPTION
**Pull request by macobo**
Another issue plaguing team deletion was that django ORM loads all
objects for a team into memory. If a team had >200k people, that would
cause the process to fail.

We now instead delete via a single sql statement.

https://github.com/PostHog/posthog/issues/10132

### CODE REVIEW

File: posthog/models/team/util.py [hunk 1]
```diff
@@ -17,10 +18,23 @@
...
+def delete_bulky_postgres_data(team_ids: List[int]):
+    "Efficiently delete large tables for teams from postgres. Using normal CASCADE delete here can time out"
+
+    _raw_delete(PersonDistinctId.objects.filter(team_id=team_ids))
+    _raw_delete(Person.objects.filter(team_id=team_ids))
+
+
+def _raw_delete(queryset: Any):
+    "Issues a single DELETE statement for the queryset"
+    queryset._raw_delete(queryset.db)
```
Comment (EDsCODE) [inline posthog/models/team/util.py:34]: 🙌

**Review by EDsCODE — APPROVED** [review]
I have a raw_delete used in deleting cohortpeople so replacing with this util will be nice

File: posthog/models/team/util.py [hunk 2]
```diff
@@ -17,10 +18,22 @@
...
     "cohortpeople",
     "person_static_cohort",
+    "plugin_log_entries",
 ]
 
 
-def delete_teams_data(team_ids: List[int]):
+def delete_bulky_postgres_data(team_ids: List[int]):
+    "Efficiently delete large tables for teams from postgres. Using normal CASCADE delete here can time out"
+    _raw_delete(PersonDistinctId.objects.filter(team_id__in=team_ids))
```
Comment (liyiy) [inline posthog/models/team/util.py:27]: @EDsCODE merge if you think this fix looks correct 😃

**Comment by macobo** [comment]
Please don't merge - I had several learnings/follow-ups from trying to delete a large client on friday.

**Pull request merged**